### PR TITLE
Update `boot-cw-plus`

### PR DESCRIPTION
**Pull request by CyberHoward**
Rehaul the boot-cw-plus crate. 

It now uses the [abstract fork of cw-plus](https://github.com/Abstract-OS/cw-plus) which contains the derive macro's for generating the interface functions. 

A top-level `CwPlus` object exposes the `store_on` function to easily upload CW plus to any mock or daemon. 
An included executable is used to fetch the `.wasm` files for each of the contracts from the cw-plus repo. 

Small adjustments were made to the function derive marco to support `ExecuteMsg`s with an "execute" variant. 

We now also re-export the ibc-registry types required to easily query chain information when the `daemon` feature is enabled.

### CODE REVIEW

File: boot-cw-plus/cw-artifacts/cw1_subkeys.wasm
Comment (adairrr) [inline boot-cw-plus/cw-artifacts/cw1_subkeys.wasm:1]: Perhaps we should include these in a folder with the version of the cw-plus artifacts.

File: boot-cw-plus/src/cw_plus.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    pub cw20_base: Cw20Base<Chain>,
+    pub cw20_ics20: Cw20Ics20<Chain>,
+}
+
+impl<Chain: CwEnv> Deploy<Chain> for CwPlus<Chain> {
+    type Error = BootError;
+    type DeployData = Empty;
+
+    /// Deploy all cw-plus contracts to the given chain.
+    fn deploy_on(chain: Chain, _data: Empty) -> Result<Self, BootError> {
```
Comment (adairrr) [inline boot-cw-plus/src/cw_plus.rs:21]: Perhaps `upload_to` would be more appropriate. `deploy` implies that it is instantiated as well.

File: boot-cw-plus/src/contracts/cw1_subkeys.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+use boot_core::{contract, Contract, CwEnv};
+use cosmwasm_std::Empty;
+pub use cw1_subkeys::msg::{
+    ExecuteMsgFns as Cw1SubkeysExecuteMsgFns, QueryMsgFns as Cw1SubkeysQueryMsgFns,
+};
+use cw1_whitelist::msg::{ExecuteMsg, InstantiateMsg, QueryMsg};
```
Comment (adairrr) [inline boot-cw-plus/src/contracts/cw1_subkeys.rs:6]: This should probably be `cw1_subkeys::msg`

File: boot-cw-plus/examples/cw-plus-mock.rs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+use boot_core::*;
+use boot_cw_plus::{Cw20QueryMsgFns, CwPlus};
+use cosmwasm_std::Addr;
+use cw20::Cw20Coin;
+
+// shows how to deploy CwPlus to a mock environment
```
Comment (adairrr) [inline boot-cw-plus/examples/cw-plus-mock.rs:6]: nit: these main functions should probably use `///` doc comment

File: boot-cw-plus/README.md [hunk 4]
```diff
@@ -1,10 +1,28 @@
...
 
-Currently supports
-- CW-20
+## Usage
+
+add this to your `Cargo.toml`:
+
+```toml
+[dependencies]
+boot-cw-plus = { git = "" }
```
Comment (adairrr) [inline boot-cw-plus/README.md:11]: `boot-cw-plus = { git = "https://github.com/Abstract-OS/BOOT" }`?

File: boot-cw-plus/src/contracts/cw1_subkeys.rs (hunk 2 above)
Comment (CyberHoward) [reply boot-cw-plus/src/contracts/cw1_subkeys.rs:6]: nice spot

File: boot-cw-plus/examples/cw-plus-mock.rs (hunk 3 above)
Comment (CyberHoward) [reply boot-cw-plus/examples/cw-plus-mock.rs:6]: examples can't be imported so not something that's done 
i don't even think they are included when you run `cargo docs`

File: boot-cw-plus/src/cw_plus.rs (hunk 1 above)
Comment (CyberHoward) [reply boot-cw-plus/src/cw_plus.rs:21]: In this specific case yes, not arguing that. I'm saying that we'd need to change the trait to reflect that in for it to be true in this specific case while for all other users `deploy_on` makes more sense (look at the WynDex integration and our own Abstract implementation of this trait)

**Comment by CyberHoward** [comment]
If you want I can add a function to the trait `upload_to`

File: boot-cw-plus/src/cw_plus.rs (hunk 1 above)
Comment (CyberHoward) [reply boot-cw-plus/src/cw_plus.rs:21]: will add an `upload_to` fn to the trait

File: boot-cw-plus/cw-artifacts/cw1_subkeys.wasm
Comment (CyberHoward) [reply boot-cw-plus/cw-artifacts/cw1_subkeys.wasm:1]: Yeah if someone requests it we can, people shouldn't use these WASMs in production.

**Pull request merged**